blog/parser.ts

```diff
@@ -0,0 +1,44 @@
+import { bundleMDX } from "mdx-bundler";
+import type { BundleMDXSource } from "mdx-bundler/dist/types";
+import type { PostFrontMatter } from "./models";
+/**
+ * Not super proud of this ESM workaround, but it's the only thing
+ * that would allow images to be used in blog posts.
+ */
+const remarkMdxImages = require("fix-esm").require("remark-mdx-images");
+
+export type Mdx = Awaited<ReturnType<typeof parseMdx>>;
+
+export async function parseMdx(
+  content: string,
+  slug: string,
+  cwd: string,
+  files?: BundleMDXSource<PostFrontMatter>["files"]
+) {
+  const { code, frontmatter } = await bundleMDX<PostFrontMatter>({
+    source: content,
+    files,
+    cwd,
```

Comment:
i don't think i realized that `bundleMDX` took a `cwd`. i _think_ if you supply the right `cwd`, you might not need to provide the comp files since they'll be resolved themselves.

however, i'm curious how that works in prod, where you don't necessarily have a file system.

Reply:
Holy 💩 it works! I can provide `cwd` and components resolve themselves. We can ✂️ another file as a result!

Reply:
Images too. And this works in dev and after build. You can check by running `npm run build` and `npm start` after.